pkg/resource/deploy/snapshot.go

```diff
@@ -300,17 +300,36 @@ func (snap *Snapshot) withUpdatedResources(update func(*resource.State) *resourc
 type SnapshotIntegrityError struct {
 	// The underlying error that caused this integrity error, if applicable.
 	Err error
+
+	// The operation which caused the error. Defaults to SnapshotIntegrityWrite.
+	Op SnapshotIntegrityOperation
```

Comment:
This is going to be "" in most cases because of zero init. Should we make SnapshotIntegrityWrite be "" (or 0 and using int enum instead) so the zero value is valid.

Reply:
Then again we never struct init this struct so maybe this is fine.

Reply:
I can make it an `int` enum. Really wanted the struct to be private but then your `As*` function is a bit less ergonomic so I think this is an OK compromise.